src/Resolvers/Microsoft.NET.Sdk.WorkloadManifestReader/SdkDirectoryWorkloadManifestProvider.cs

```diff
@@ -43,16 +43,28 @@ internal SdkDirectoryWorkloadManifestProvider(string sdkRootPath, string sdkVers
             _sdkVersionBand = new SdkFeatureBand(sdkVersion);
 
             var knownManifestIdsFilePath = Path.Combine(_sdkRootPath, "sdk", sdkVersion, "KnownWorkloadManifests.txt");
+            int lineNumber = 0;
+
             if (File.Exists(knownManifestIdsFilePath))
             {
                 _knownManifestIdsAndOrder = new Dictionary<string, int>();
-                int lineNumber = 0;
                 foreach (var manifestId in File.ReadAllLines(knownManifestIdsFilePath).Where(l => !string.IsNullOrEmpty(l)))
                 {
                     _knownManifestIdsAndOrder[manifestId] = lineNumber++;
                 }
             }
-
+            var includedManifestIdsFilePath = Path.Combine(_sdkRootPath, "sdk", sdkVersion, "IncludedWorkloadManifests.txt");
+            if (File.Exists(includedManifestIdsFilePath))
```

Comment:
I wouldn't duplicate the logic or ever read both files.  How about something like this:

```c#
if (!File.Exists(knownManifestIdsFilePath))
{
    knownManifestIdsFilePath = Path.Combine(_sdkRootPath, "sdk", sdkVersion, "IncludedWorkloadManifests.txt");
}

if (File.Exists(knownManifestIdsFilePath))
{
    // Existing logic to read the file}
```

Reply:
Ahh, I thought you meant read both files in case a testing environment ended up with both. I should have recognized that that didn't really make sense to read them both from the same sdk as that would only happen with copying files around.